### ISSUE TITLE
Problems with transitions

### ISSUE THREAD

**Issue by He-Zu** [OpenShot/openshot-qt#5544]
When I overlap two jpg images, a transition is created. This works, but the time is too short for me. If I change the duration of the transition, the crossfade is defective. see example video. Even if I take another transition from the selection, it doesn't work. What am I doing wrong?
Can anyone help me?
Thank you very much

https://github.com/OpenShot/openshot-qt/assets/116963345/16378ae2-7e67-439d-9340-dbd89ad06d4c

**Comment by Colorjet3** [comment]
This is a known bug and in the queue to be fixed in a future release(note:  there is a v3.1.2 in the works to be released but unfortunately this fix is not in it at this time).

A workaround for this would be:
1.  Drag the first image on Track 5.
2.  Drag the 2nd image on Track 4.
3.  Overlap them as if they were on the same track.
4.  From the Transitions tab drag the Fade transition or whichever one you want onto the clip on Track 5.  Adjust the left edge of the transition so it is only covering passed the 1st half of the clip on Track 5.  This will achieve Fade Out.
5.  Do the same for the clip on Track 4 but adjust the right edge of the Fade transition so it only covers less than the 1st half of the clip.  This will achieve Fade In.
6. Preview and you should see the transition working properly.

**Comment by eugenialoli** [comment]
This is what I'm experiencing too, on the other bug report I did. Your workaround works, but it's been 6 months and this hasn't been fixed yet. Transitions are the no1 thing people do after putting a couple of clips in the timeline. They usually don't add effects, but they do go for transitions. This bug is a show stopper imho.

**Comment by Colorjet3** [comment]
Hello @eugenialoli This project is dependent on volunteers and the lead developer just doesn't have enough resources to get fixes into production on a timely fashion.

**Comment by Colorjet3** [comment]
Hello @eugenialoli Please go to openshot.org/download/#daily and download v3.4.0-Release-Candidate.  Retest again and report back with your results.

I expect that this should be working now.

**Comment by eugenialoli** [comment]
I just tried the new version, it's 99% there. Only thing is missing is to change the Fade to another transition from the properties window. Where it says "Fade" in the drop down menu, it should have a list of ALL the transitions, so we can change the default transition to whatever we want, without having to first delete the fade transition, enter a new transition, resize the new transition, reposition the new transition. It should be part of the drop down box in the Fade properties box. 

But yes, the basic idea is fixed. When you overlap two clips, it should create a transition automatically. This now works. Only thing remaining is to change that transition just as easily to another one.

**Comment by eugenialoli** [comment]
No, I spoke too soon. If I create a new transition and put it on top of the two overlapping clips, there are many frames in the beginning that are all blacked out. However, if I take one of the clips on the bottom track, and then put it back up, then the issue disappears. So this needs more work to work as expected.

**Comment by Colorjet3** [comment]
Thank you @eugenialoli 

You don't have to delete the Fade transition if you want to change it to another transition.  Here is what you do:
1.  Click the Fade transition and right click to see the Properties.
2.  Find the Source attribute, right click the empty space to the right and click on Transitions.
3.  Now you can select another transition.
4.  This works even if you made adjustments to the original Fade transition.

If this isn't what you are trying to convey then please provide me with step-by-step instructions and for each step explain what you expected to happen.

As for the "blacked out" frames.  By default, when you add a transition to a clip and it covers the entire clip, by default all transitions Fade in and Fade out.

If I am misunderstanding, please provide me with step-by-step instructions so I can follow it to replicate your issue.

Also, the step-by-step instructions to reproduce an issue is what the lead developer looks for when troubleshooting.

Thank you for your time.

**Comment by eugenialoli** [comment]
>As for the "blacked out" frames. By default, when you add a transition to a clip and it covers the entire clip, by default all transitions Fade in and Fade out.

It doesn't cover all of the clip, it covers only the overlapping of the two clips (where the "fade" used to be before I deleted the transition and added another one). It should not create black frames.

**Comment by Colorjet3** [comment]
You may be running into an existing bug (very difficult to reproduce, that happens even without transitions).  When this happens, please select all your clips on a track, shift them to the right by say 10 seconds.  Let the Cache rebuild.  Now jumpt the playhead to the first clip.

Do you see your frames now?  If not, grab all the clips again on that track and shift them left and right and let go.  Doing this might/will display the blackened frames (get rid of the blackness).

You can now shift all the clips back to position 0:00 and see if all is good.

**Comment by eugenialoli** [comment]
Yes, I've actually noticed that when I move one of the two overlapping clips to another track, and then put it back where it was, the black frames disappear. This needs to be fixed asap imho, because transitions is the no1 thing people do after placing a few clips in a timeline.

**Comment by Colorjet3** [comment]
I agree and it is in the queue.  However, it is all up to the lead developer to decide which ones he has time to work on.  All I do is report the issues/enhancements.

Besides this bug, are you all good with the Transitions now based on our thread?
